### PR TITLE
storage/file: Preliminary support for Azure File Service

### PR DESCRIPTION
**Pull request by ahmetalpbalkan**
Support for CreateShare/DeleteShare calls.

@paulmey PTAL.

### CODE REVIEW

File: storage/blob_test.go [hunk 1]
```diff
@@ -168,7 +168,7 @@ func (s *StorageBlobSuite) TestContainerExists(c *chk.C) {
 	c.Assert(ok, chk.Equals, true)
 }
 
-func (s *StorageBlobSuite) TestCreateDeleteContainer(c *chk.C) {
+func (s *StorageBlobSuite) TestCreateContainerDeleteContainer(c *chk.C) {
```
Comment (paulmey) [inline storage/blob_test.go]: Sneaking in some test fixes eh? Well, ok then, but only for this time...

Comment (ahmetalpbalkan) [inline storage/blob_test.go]: I ctrl-f'ed TestCreateContainer and wondered how come I missed this one, turns out it was just naming thing. Better searchability fix.

File: storage/file_test.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	defer cli.DeleteShare(name)
+
+	// First create
+	ok, err := cli.CreateShareIfNotExists(name)
+	c.Assert(err, chk.IsNil)
+	c.Assert(ok, chk.Equals, true)
+
+	// Second create, should not give errors
+	ok, err = cli.CreateShareIfNotExists(name)
+	c.Assert(err, chk.IsNil)
```
Comment (paulmey) [inline storage/file_test.go]: do you also want to check the value of `ok`?

Comment (ahmetalpbalkan) [inline storage/file_test.go]: good catch

Comment (ahmetalpbalkan) [inline storage/file_test.go]: fixed

**Comment by paulmey** [comment]
LGTM, but the build is failing. Looks like it may not be calculating the `file` endpoint correctly?

**Comment by ahmetalpbalkan** [comment]
nope it used to pass in the earlier commit... I believe it is a network issue in Travis. Kicked off a rebuild.

**Pull request merged**